src/main/java/saps/archiver/core/DefaultPermanentStorage.java

```diff
@@ -1,15 +1,70 @@
+/* (C)2020 */
 package saps.archiver.core;
 
-import saps.archiver.interfaces.PermanentStorageInterface;
+import static saps.common.core.storage.PermanentStorageConstants.INPUTDOWNLOADING_DIR;
+import static saps.common.core.storage.PermanentStorageConstants.PREPROCESSING_DIR;
+import static saps.common.core.storage.PermanentStorageConstants.PROCESSING_DIR;
+import static saps.common.core.storage.PermanentStorageConstants.SAPS_TASK_STAGE_DIR_PATTERN;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Properties;
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
 import saps.common.core.model.SapsImage;
+import saps.archiver.interfaces.PermanentStorage;
+import saps.common.core.storage.exceptions.InvalidPropertyException;
+import saps.common.utils.SapsPropertiesConstants;
+
+public class FSPermanentStorage implements PermanentStorage {
+
+  public static final Logger LOGGER = Logger.getLogger(FSPermanentStorage.class);
+  public static final String NFS_STORAGE_TASK_DIR_PATTERN ="%s" + File.separator + "%s" + File.separator + "%s";
+  public static final String NFS_STORAGE_TASK_URL_PATTERN = "%s" + File.separator + "%s";
+  
+  private final String nfsTempStoragePath;
+  private final String nfsPermanentStoragePath;
+  private final String tasksDirName;
+
+  public FSPermanentStorage(Properties properties) throws InvalidPropertyException {
+    this.nfsTempStoragePath = properties.getProperty(SapsPropertiesConstants.SAPS_TEMP_STORAGE_PATH);
+    this.nfsPermanentStoragePath = properties.getProperty(SapsPropertiesConstants.NFS_PERMANENT_STORAGE_PATH);
+    this.tasksDirName = properties.getProperty(SapsPropertiesConstants.PERMANENT_STORAGE_TASKS_DIR); 
+  }
+
+  @Override
```

Comment:
rename everything that is named NFS to FS, including variables and properties. I think it is importante because this implementation does not depend on NFS. It works for any file system